pkg/core/native/contract.go

```diff
@@ -111,12 +112,15 @@ func NewContracts(p2pSigExtensionsEnabled bool, nativeUpdateHistories map[string
 		cs.Contracts = append(cs.Contracts, notary)
 	}
 
-	setDefaultHistory := len(nativeUpdateHistories) == 0
+	histories := make(map[string][]uint32)
+	if len(cfg.NativeUpdateHistories) != 0 {
+		histories = cfg.NativeUpdateHistories
+	}
 	for _, c := range cs.Contracts {
-		if setDefaultHistory {
-			nativeUpdateHistories[c.Metadata().Name] = []uint32{0}
+		if histories[c.Metadata().Name] == nil {
```

Comment:
Consider the case when several (but not all) histories are set in the config. Then `nil` history means that the contract is completely disabled (inactive). But this code sets the history of such disabled contract to `[]uint32{0}` which makes the contract enabled.

So if I'm not mistaken, we should set the defaults only when **all** histories are not specified in the config, that's how the reference implementation works.

Reply:
Fixed.